loopd/daemon.go

```diff
@@ -259,6 +229,58 @@ func (d *Daemon) Run() error {
 	return nil
 }
 
+// initializeClient creates and initializes an instance of the swap server
+// client.
+func (d *Daemon) initializeClient() error {
+	// If no swap server is specified, use the default addresses for mainnet
+	// and testnet.
+	if d.cfg.SwapServer == "" {
+		// TODO(wilmer): Use onion service addresses when proxy is
+		// active.
+		switch d.cfg.Network {
+		case "mainnet":
+			d.cfg.SwapServer = mainnetServer
+		case "testnet":
+			d.cfg.SwapServer = testnetServer
+		default:
+			return errors.New("no swap server address specified")
+		}
+	}
+
+	log.Infof("Swap server address: %v", d.cfg.SwapServer)
+
+	// Create an instance of the loop client library.
+	var err error
+	d.impl, d.clientCleanup, err = getClient(d.cfg, &d.lnd.LndServices)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve all currently existing swaps from the database.
+	swapsList, err := d.impl.FetchSwaps()
+	if err != nil {
+		return err
```

Comment:
is `clientCleanup` called in this case?

Reply:
Same for other errors that may happen in `Start`

Reply:
Yes, you're right. I added a commit that calls `d.Stop()` before returning an error. Because everything in `Stop()` is guarded against being nil, we can do that.

Reply:
But in this commit, it is incorrect? I think the commits should be logical units of change that don't leave open ends

Reply:
I've added the stop to the commit where it's first introduced. I also created the `stopAndReturnErr` helper.